### PR TITLE
add sample transformation

### PR DESCRIPTION
**Pull request by seanzhougoogle**

### CODE REVIEW

File: src/main/java/io/cdap/delta/transformation/MaskTransformation.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      throw new IllegalArgumentException("Directive should have 4 arguments. Usage: mask column_name from_position " +
+                                           "to_position.");
+    }
+    if (!NAME.equals(splits[0])) {
+      throw new IllegalArgumentException("Directive is not a mask transformation. Usage: mask column_name " +
+                                           "from_position to_position.");
+    }
+
+    srcColumn = splits[1];
+    fromPos = Integer.parseInt(splits[2]);
```
Comment (sagarkapare) [inline src/main/java/io/cdap/delta/transformation/MaskTransformation.java]: maybe better to catch NumberFormatException too and return appropriate error message while throwing IllegalArgumentException

File: src/main/java/io/cdap/delta/transformation/MaskTransformation.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    toPos = Integer.parseInt(splits[3]);
```
Comment (sagarkapare) [inline src/main/java/io/cdap/delta/transformation/MaskTransformation.java]: same here.

File: src/main/java/io/cdap/delta/transformation/MaskTransformation.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    if (fromPos > toPos) {
+      throw new IllegalArgumentException("to_position is greater than from_position.");
+    }
+  }
+
+  @Override
+  public void transformValue(MutableRowValue rowValue) throws Exception {
+    String value;
+    try {
+      value = (String) rowValue.getColumnValue(srcColumn);
```
Comment (sagarkapare) [inline src/main/java/io/cdap/delta/transformation/MaskTransformation.java:83]: Will this only be apply to String types? Will be good to confirm from product.

**Review by sagarkapare — APPROVED** [review]
minor questions. else LGTM 👍

File: src/main/java/io/cdap/delta/transformation/MaskTransformation.java (hunk 2 above)
Comment (seanzhougoogle) [reply src/main/java/io/cdap/delta/transformation/MaskTransformation.java]: done :)

File: src/main/java/io/cdap/delta/transformation/MaskTransformation.java (hunk 1 above)
Comment (seanzhougoogle) [reply src/main/java/io/cdap/delta/transformation/MaskTransformation.java]: done :)

File: src/main/java/io/cdap/delta/transformation/MaskTransformation.java (hunk 3 above)
Comment (seanzhougoogle) [reply src/main/java/io/cdap/delta/transformation/MaskTransformation.java:83]: this is just some demo code, for 6.6. we haven't got clear requirement for mask transformation. Will double check with product when we do 6.6 design.

**Pull request merged**